### PR TITLE
Query progress widget [API-1445]

### PR DESCRIPTION
**Pull request by utku-caglayan**
Adds a widget to SQL browser to indicate the progress on a query

### CODE REVIEW

File: internal/browser/seperator.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const (
+	HideProgress = iota
+	ShowProgress
+)
+
+var spinnerWidget = spinner.New()
+
+func changeProgress(ps int32) {
+	switch {
+	case ps == HideProgress:
```
Comment (yuce) [inline internal/browser/seperator.go:24]: Why not:
```
switch ps {
    case HideProgress:
// ...
```

File: internal/browser/seperator.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	length int
+}
+
+func (s *SeparatorWithProgress) Init() tea.Cmd {
+	return spinnerWidget.Tick
+}
+
+func (s *SeparatorWithProgress) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	if msg, ok := msg.(tea.WindowSizeMsg); ok {
+		s.length = msg.Width - 2
```
Comment (yuce) [inline internal/browser/seperator.go:45]: What is `-2`? Does it make sense to move it to a named constant?

File: internal/browser/seperator.go (hunk 1 above)
Comment (utku-caglayan) [reply internal/browser/seperator.go]: that's better, thanks! https://github.com/hazelcast/hazelcast-commandline-client/pull/104/commits/925164c7c36af50ed95e949a2686edda2bce66f2

File: internal/browser/seperator.go (hunk 2 above)
Comment (utku-caglayan) [reply internal/browser/seperator.go]: Removed it completely https://github.com/hazelcast/hazelcast-commandline-client/pull/104/commits/925164c7c36af50ed95e949a2686edda2bce66f2

**Review by yuce — APPROVED** [review]
Looks good!

**Pull request merged**